src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContext.cs

```diff
@@ -11,13 +11,15 @@ namespace NuGet.VisualStudio
 {
     // Implementation of IVsPathContext without support of reference resolving.
     // Used when project is not managed by NuGet or given project doesn't have any packages installed.
-    internal class VsPathContext : IVsPathContext
+    internal class VsPathContext : IVsPathContext2
     {
         public string UserPackageFolder { get; }
 
         public IEnumerable FallbackPackageFolders { get; }
 
-        public VsPathContext(NuGetPathContext pathContext)
+        public string SolutionPackageFolder { get; }
+
+        public VsPathContext(NuGetPathContext pathContext, string solutionPackageFolder = null)
```

Comment:
nit: consider having two separate constructors with and without `solutionPackageFolder` instead of using a default of `null`

Reply:
actually I like this better :) 
I prefer optional parameters like this instead of (potentially) 5 different constructors.

Reply:
either way doesn't make any difference so I'll accept your suggestion :)

Reply:
In C# I prefer using polymorphism and have multiple methods instead of one with optional parameters because it works better with Intellisense in VS (it has the ability to give you more descriptive information about each parameter and what are your options when calling the method). Also, this is the style we follow everywhere else in the product.

Reply:
IMO intellisense works similar in both ways, even with optional parameter, it clearly shows you all the options with default value if you don't specify. Even I agree with Nikolche's point of having single constructor instead of multiple variations. 

`Also, this is the style we follow everywhere else in the product.`

This is not entirely true, we have both styles being followed (depending on the engineer, I guess) like check `NUGetVersion` or `SemanticVersion`, etc...